libpod/container_internal.go

```diff
@@ -1125,57 +1125,11 @@ func (c *Container) saveSpec(spec *spec.Spec) error {
 	return nil
 }
 
-func (c *Container) setupOCIHooks(g *generate.Generator) error {
-	addedHooks := map[string]struct{}{}
-	ocihooks, err := hooks.SetupHooks(c.runtime.config.HooksDir)
+func (c *Container) setupOCIHooks(ctx context.Context, g *generate.Generator) error {
+	manager, err := hooks.New(ctx, []string{c.runtime.config.HooksDir})
 	if err != nil {
 		return err
 	}
-	addHook := func(hook hooks.HookParams) error {
-		// Only add a hook once
-		if _, ok := addedHooks[hook.Hook]; !ok {
-			if err := hooks.AddOCIHook(g, hook); err != nil {
-				return err
-			}
-			addedHooks[hook.Hook] = struct{}{}
-		}
-		return nil
-	}
-	for _, hook := range ocihooks {
-		logrus.Debugf("SetupOCIHooks", hook)
-		if hook.HasBindMounts && len(c.config.Spec.Mounts) > 0 {
-			if err := addHook(hook); err != nil {
-				return err
-			}
-			continue
-		}
-		for _, cmd := range hook.Cmds {
-			match, err := regexp.MatchString(cmd, c.config.Spec.Process.Args[0])
-			if err != nil {
-				logrus.Errorf("Invalid regex %q:%q", cmd, err)
-				continue
-			}
-			if match {
-				if err := addHook(hook); err != nil {
-					return err
-				}
-			}
-		}
-		annotations := c.Spec().Annotations
-		for _, annotationRegex := range hook.Annotations {
-			for _, annotation := range annotations {
-				match, err := regexp.MatchString(annotationRegex, annotation)
-				if err != nil {
-					logrus.Errorf("Invalid regex %q:%q", annotationRegex, err)
-					continue
-				}
-				if match {
-					if err := addHook(hook); err != nil {
-						return err
-					}
-				}
-			}
-		}
-	}
-	return nil
+	hasBindMounts := false  // FIXME: what do we want here?  https://github.com/projectatomic/libpod/pull/155#pullrequestreview-116002429
```

Comment:
I think this can probably be `len(c.config.Spec.Mounts) > 0`... It's not 1:1 with actual bind mounts, but it's the closest we can get inside libpod

Reply:
> I think this can probably be `len(c.config.Spec.Mounts) > 0`... It's not 1:1 with actual bind mounts, but it's the closest we can get inside libpod

We can always adjust the libpod API to be more precise.  However, you're suggesting we just keep the current logic, so I expect we should punt any libpod API changes to follow-up work if we address this issue at all.

Can you recommend wording for the [the docs][1] explaining the difference, so we document the behavior to avoid surprising users?

[1]: https://github.com/projectatomic/libpod/pull/686/files#diff-6ff75d2b4fc02ea5ead9bd97530d6f2aR42

Reply:
On further examination, we do add several mounts unconditionally to the spec by default, so my change compiles into an unconditional `true`, avoiding the actual intent of the hook...

Given this, I think it's safest right now to keep your unconditional `false` and document that hasBindMounts hooks are not currently implemented. We could add a boolean to config indicating that user-added bind mounts are present for the purpose of informing the hooks at a later date.

Reply:
> Given this, I think it's safest right now to keep your unconditional `false` and document that `hasBindMounts` hooks are not currently implemented.

Done and pushed as 51fb6c48.

Reply:
oci-umount relies on this.  So it would be better if you unconditionally return true, until it is fixed.  I would rather always run the hook and take the penalty rather then face problems with leaked mount points.  But we need to fix this quickly.

Reply:
> But we need to fix this quickly.

It's probably better if @mheon or someone else familiar with the config flow files a separate PR to update libpod to handle this correctly.  I'll rebase this on top once that lands.

Reply:
@wking Sure, should be simple enough. I'll see about getting one in on Monday.